src/snapred/ui/workflow/WorkflowImplementer.py

```diff
@@ -50,6 +50,11 @@ def __init__(self, parent=None):
         self.continueAnywayFlags = ContinueWarning.Type.UNSET
         self.responseHandler = SNAPResponseHandler(self.parent)
 
+        self.resetHooks = []
+
+    def addResetHook(self, hook):
```

Comment:
`def addResetHook(self, hook: Callable[[], None]):`  : I think it's always important to specify the expected signature and return value of any function passed as an argument.